CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+# Minimum CMake required
+cmake_minimum_required(VERSION 3.1)
+
+# Project
+project(onnxir C CXX)
+
+# Set C++11 as standard for the whole project
+set(CMAKE_CXX_STANDARD 11)
+
+set(ONNXIR_ROOT ${PROJECT_SOURCE_DIR})
+if (WIN32)
+    # parallel build
+    add_compile_options(/MP)
+    add_compile_options(/WX)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WX")
+    set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} /WX")
+endif()
+add_subdirectory(third_party/pybind11)
+FIND_PACKAGE(Protobuf REQUIRED)
+include_directories(
+    ${ONNXIR_ROOT}
+    ${CMAKE_CURRENT_BINARY_DIR}
+    # External dependencies.
+    ${PROTOBUF_INCLUDE_DIRS}
+    ${PYBIND11_INCLUDE_DIR}
+    ${PYTHON_INCLUDE_DIRS}
+)
+
+function(RELATIVE_PROTOBUF_GENERATE_CPP SRCS HDRS ROOT_DIR)
+  if(NOT ARGN)
+    message(SEND_ERROR "Error: RELATIVE_PROTOBUF_GENERATE_CPP() called without any proto files")
+    return()
+  endif()
+
+  set(${SRCS})
+  set(${HDRS})
+  foreach(FIL ${ARGN})
+    message(${FIL})
+    set(ABS_FIL ${ROOT_DIR}/${FIL})
+    get_filename_component(FIL_WE ${FIL} NAME_WE)
+    get_filename_component(FIL_DIR ${ABS_FIL} DIRECTORY)
+    file(RELATIVE_PATH REL_DIR ${ROOT_DIR} ${FIL_DIR})
+
+    list(APPEND ${SRCS} "${CMAKE_CURRENT_BINARY_DIR}/${REL_DIR}/${FIL_WE}.pb.cc")
+    list(APPEND ${HDRS} "${CMAKE_CURRENT_BINARY_DIR}/${REL_DIR}/${FIL_WE}.pb.h")
+
+    add_custom_command(
+      OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${REL_DIR}/${FIL_WE}.pb.cc"
+             "${CMAKE_CURRENT_BINARY_DIR}/${REL_DIR}/${FIL_WE}.pb.h"
+      COMMAND ${PROTOBUF_PROTOC_EXECUTABLE}
+      ARGS --cpp_out  ${CMAKE_CURRENT_BINARY_DIR} -I ${ROOT_DIR} ${ABS_FIL} -I ${PROTOBUF_INCLUDE_DIRS} -I ${ROOT_DIR}/${REL_DIR}
+      DEPENDS ${ABS_FIL} protobuf
+      COMMENT "Running C++ protocol buffer compiler on ${FIL}"
+      VERBATIM )
+  endforeach()
+
+  set_source_files_properties(${${SRCS}} ${${HDRS}} PROPERTIES GENERATED TRUE)
+  set(${SRCS} ${${SRCS}} PARENT_SCOPE)
+  set(${HDRS} ${${HDRS}} PARENT_SCOPE)
+endfunction()
+
+RELATIVE_PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS ${ONNXIR_ROOT}
+    onnx/onnx.proto
+)
+
+file(GLOB_RECURSE onnx_src
+    "${ONNXIR_ROOT}/onnx/*.h"
+    "${ONNXIR_ROOT}/onnx/*.cc"
+)
+
+file(GLOB_RECURSE onnx_exclude_src
```

Comment:
You don't actually need a Glob here, right?

Reply:
I think I don't need recursive here. Anyway, I'm changing to use "Remove" directly. This piece of codes will be removed anyway later.

Reply:
After changing setup.py you will still need this. We will only use setup.py to manage the python packaging part, while cmake is going to be the one that does all the heavy lifting (including invoking protoc).